corehq/ex-submodules/dimagi/utils/web.py

```diff
@@ -117,14 +119,17 @@ def json_handler(obj):
         return float(obj) # warning, potential loss of precision
     elif isinstance(obj, Promise):
         return force_text(obj)  # to support ugettext_lazy
+    elif isinstance(obj, bytes):
+        return obj.decode('utf-8')
     else:
         return json.JSONEncoder().default(obj)
 
 
 def json_response(obj, status_code=200, **kwargs):
-    """
-    Deprecated. When possible use Django's built in JsonResponse
-    """
+    warnings.warn(
+        "json_response is deprecated.  Use django.http.JsonResponse instead.",
```

Comment:
is it worth (or possible) changing HttpResponse to JsonResponse in the return? we have a habit of ignoring deprecation warnings until the last minute

Reply:
What would be the benefit?

Reply:
depends on why its deprecated. If it is deprecated because it doesn't handle json well and we're writing our own json handler, can we proactively change this to using jsonresponse to get rid of these bugs in the future?

if its deprecated because these two functions have different outputs and we prefer one output over the other, then its probably not worth/possilbe to change

Reply:
Is there a plan to eliminate this in the longer term? It does seem like it could (should?) be spun off as a separate effort from the Python 3 upgrade. Maybe add it to the tech-debt trello if nothing else?

Reply:
The original deprecation comment was added here:
https://github.com/dimagi/commcare-hq/pull/18324/commits/6480fcd570a9cbcf4f398860113a3d99fac9ea84

I think the point is that we can often use ```JsonResponse``` instead of ```json_response``` and it's preferable to use django than to support our own way of doing this.  So unless there are exceptions we should be trying to reduce (or at least not add to) the number of times ```json_response``` is used.

Reply:
https://github.com/dimagi/commcare-hq/pull/24086